openedx/core/djangoapps/discussions/transformers.py

```diff
@@ -0,0 +1,49 @@
+"""
+Discussions Topic Link Transformer
+"""
+
+from openedx.core.djangoapps.content.block_structure.transformer import BlockStructureTransformer
+from openedx.core.djangoapps.discussions.models import DiscussionTopicLink, DiscussionsConfiguration
+from openedx.core.djangoapps.discussions.url_helpers import get_discussions_mfe_topic_url
+
+
+class DiscussionsTopicLinkTransformer(BlockStructureTransformer):
+    """
+    A transformer that adds discussion topic context to the xblock.
```

Comment:
How often this transfer is called? is it per xblock/block or per course? I ask this because in the body the method is transforming all the course topics for a given provider.

Reply:
It is called once for the entire block structure. The block structure for the course is cached and each transformer can iterate over the blocks to enhance the data or remove blocks from the strucutre.

Reply:
Got it, thanks for the explaination. appriciate that.